buildSrc/src/main/java/org/opensearch/gradle/pluginzip/Publish.java

```diff
@@ -42,24 +42,7 @@ public static void configMaven(Project project) {
             });
             publishing.publications(publications -> {
                 MavenPublication mavenZip = (MavenPublication) publications.findByName(PUBLICATION_NAME);
```

Comment:
I don't think removing this logic is right move, if the plugin has no publishing section, you end up with NPE:

```
Caused by: java.lang.NullPointerException                                                                                                                                                                                                                                                                                                                                                   
        at org.opensearch.gradle.pluginzip.Publish.lambda$configMaven$2(Publish.java:45)                                                                                                                                                                                                                                                                                                    
        at org.gradle.api.publish.internal.DefaultPublishingExtension.publications(Defaul
```

Reply:
@reta Good catch. I think I can recreate.

But the question then is if there are no publication tasks then why we would need to call the `configMaven()` method at all? And it also injects the local mvn staging repo into repositories, which should not happen as well IMO.

Reply:
`Nebula` plugin may be?

Reply:
@reta We do it, check the Publish.java class, the `apply()` method (which also calls the `configMaven()`).

Reply:
I think the solution should be that if there is no `pluginZip` publication defined then the `apply()` method should skip all the initialization. That is what I am actually trying to implement as we speak...

Reply:
@reta I fixed this issue in the recent PR udpate.